phoenix-core/src/main/java/org/apache/phoenix/compile/WhereOptimizer.java

```diff
@@ -1722,7 +1725,10 @@ public KeySlots visitLeave(LikeExpression node, List<KeySlots> childParts) {
                 lowerRange = Arrays.copyOf(lowerRange, lowerRange.length+1);
                 lowerRange[lowerRange.length-1] = QueryConstants.SEPARATOR_BYTE;
             }
-            KeyRange range = type.getKeyRange(lowerRange, true, upperRange, false, SortOrder.ASC);
+            boolean lowerInclusive = column.getSortOrder() == SortOrder.ASC;
+            boolean upperInclusive = column.getSortOrder() == SortOrder.DESC;
+            KeyRange range = type.getKeyRange(lowerRange, lowerInclusive, upperRange,
+                    upperInclusive, SortOrder.ASC);
```

Comment:
I have debugged into this, and after inverting twice, we end up returning a non-inverted range, which eventually gets inverted in WhereOptimizer.pushKeyExpressionsToScan().

Do we even need to take SortOrder into account here, and if yes, couldn't we simplify this logic ?

Reply:
I'm talking the whole method, not the specific line above.